crates/sui-types/src/messages.rs

```diff
@@ -3052,6 +3070,28 @@ impl TransactionEffects {
             effects: self.digest(),
         }
     }
+
+    pub fn estimate_effects_size_upperbound(
+        num_writes: usize,
+        num_deletes: usize,
+        num_deps: usize,
+    ) -> usize {
+        // TODO: size_of is not guaranteed to be stable across platforms. We should use a fixed constant size here?
+        let fixed_sizes = mem::size_of::<ExecutionStatus>()
```

Comment:
How much of a concern is this size_of potential cross-platform instability?

Reply:
If this is used to determine whether to return with ExecutionError, then it is a serious concern because it can lead to forks?
btw can we just limit the total length of all the arrays in effects, instead of limiting the size which is expensive to get accurately?

Reply:
I actually changed this to use constants. Forgot to update the comment.
And I have changed the logic altogether to use an estimator instead of serialized size.